search/engine.go

```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
 	}
 }
 
+func createSourcesQuery(q Query) elastic.Query {
+	query := elastic.NewBoolQuery()
+	if q.Term != "" {
+		query = query.Must(
+			// Don't calculate score here, as we use sloped score below.
+			elastic.NewConstantScoreQuery(
+				elastic.NewBoolQuery().Should(
+					elastic.NewMatchQuery("name.analyzed", q.Term),
+					elastic.NewMatchQuery("description.analyzed", q.Term),
+					elastic.NewMatchQuery("content.analyzed", q.Term),
+					elastic.NewMatchQuery("authors.analyzed", q.Term),
+				).MinimumNumberShouldMatch(1)).Boost(1),
+		).Should(
+			elastic.NewBoolQuery().Should(
+				elastic.NewMatchPhraseQuery("name.analyzed", q.Term).Slop(100).Boost(1.5),
+				elastic.NewMatchPhraseQuery("description.analyzed", q.Term).Slop(100).Boost(1.2),
+				elastic.NewMatchPhraseQuery("content.analyzed", q.Term).Slop(100),
+				elastic.NewMatchPhraseQuery("authors.analyzed", q.Term).Slop(100),
+			).MinimumNumberShouldMatch(0),
+		)
+	}
+	for _, exactTerm := range q.ExactTerms {
+		query = query.Must(
+			// Don't calculate score here, as we use sloped score below.
+			elastic.NewConstantScoreQuery(
+				elastic.NewBoolQuery().Should(
+					elastic.NewMatchPhraseQuery("name", exactTerm),
+					elastic.NewMatchPhraseQuery("description", exactTerm),
+					elastic.NewMatchPhraseQuery("content", exactTerm),
+					elastic.NewMatchPhraseQuery("authors", exactTerm),
+				).MinimumNumberShouldMatch(1)).Boost(1),
+		).Should(
+			elastic.NewBoolQuery().Should(
+				elastic.NewMatchPhraseQuery("name", exactTerm).Slop(100).Boost(1.5),
+				elastic.NewMatchPhraseQuery("description", exactTerm).Slop(100).Boost(1.2),
+				elastic.NewMatchPhraseQuery("content.analyzed", exactTerm).Slop(100),
+				elastic.NewMatchPhraseQuery("authors.analyzed", exactTerm).Slop(100),
+			).MinimumNumberShouldMatch(1),
+		)
+	}
+	return elastic.NewFunctionScoreQuery().Query(query)
+	//.AddScoreFunc(elastic.NewGaussDecayFunction().FieldName("effective_date").Decay(0.9).Scale("300d"))
+}
+
+func AddSourcesSearchRequests(mss *elastic.MultiSearchService, query Query, from int, size int, preference string) {
+	sources_indices := make([]string, len(query.LanguageOrder))
+	for i := range query.LanguageOrder {
+		sources_indices[i] = es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i])
+		log.Infof("adding index to sources_indices: %s", es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i]))
+	}
+	fetchSourceContext := elastic.NewFetchSourceContext(true).
+		Include("mdb_uid")
+	for _, index := range sources_indices {
+		searchSource := elastic.NewSearchSource().
+			Query(createSourcesQuery(query)).
+			Highlight(elastic.NewHighlight().Fields(
```

Comment:
Should be `Highlight(elastic.NewHighlight().HighlighterType("unified").Fields(`

Reply:
done